src/main/scala/io/getclump/ClumpContext.scala

```diff
@@ -32,27 +22,27 @@ private[getclump] final class ClumpContext {
     flush(clumps.map(_.upstream).flatten)
 
   private[this] def flushDownstream(clumps: List[Clump[_]]) =
-    Future.collect(clumps.map(_.downstream)).flatMap { down =>
+    Future.sequence(clumps.map(_.downstream)).flatMap { down =>
       flush(down.flatten.toList)
     }
 
-  private[this] def flushFetches(clumps: List[Clump[_]]) =
-    Future.collect(fetchersFor(clumps).map(_.flush))
+  private[this] def flushFetches(clumps: List[Clump[_]]) = {
+    val fetches = filterFetches(clumps)
+    val byFetcher = fetches.groupBy(fetch => fetcherFor(fetch.source))
+    for ((fetcher, fetches) <- byFetcher)
+      fetches.foreach(_.attachTo(fetcher))
+    Future.sequence(byFetcher.keys.map(_.flush).toSeq)
+  }
 
-  private[this] def fetchersFor(clumps: List[Clump[_]]) =
+  private[this] def filterFetches(clumps: List[Clump[_]]) =
     clumps.collect {
-      case clump: ClumpFetch[_, _] => clump.fetcher
-    }.distinct
-}
-
-private[getclump] object ClumpContext {
-
-  private[this] val local = new Local[ClumpContext]
```

Comment:
We had to get rid of the `Local` usage because Scala Futures don't have this feature.